src/lib/prisma-session-store.ts

```diff
@@ -339,11 +337,7 @@ export class PrismaSessionStore<M extends string = 'session'> extends Store {
         const sid = session.sid;
         this.logger.log(`Deleting session with sid: ${sid}`);
         const foundSession = await p.findUnique({ where: { sid } });
-        if (foundSession !== null) {
-          await p.delete({ where: { sid } });
-        } else {
-          this.logger.warn(`Session record with sid: ${sid} not found.`);
-        }
+        if (foundSession !== null) await p.delete({ where: { sid } });
```

Comment:
There is still a bug here, since the read and the subsequent write are not in a transaction.

What about switching to `deleteMany` which should be idempotent, then we don't have to do two DB operations here or worry about concurrency...